examples/green_screen/main.cpp

```diff
@@ -0,0 +1,529 @@
+    return tr;
+}
+
+// This function constructs a calibration that operates as a transformation between the secondary device's depth camera
+// and the main camera's color camera. IT WILL NOT GENERALIZE TO OTHER TRANSFORMS. Under the hood, the transformation
+// depth_image_to_color_camera method can be thought of as converting each depth pixel to a 3d point using the
+// intrinsics of the depth camera, then using the calibration's extrinsics to convert between depth and color, then
+// using the color intrinsics to produce the point in the color camera perspective.
+k4a::calibration construct_device_to_device_calibration(const k4a::calibration &main_cal,
+                                                        const k4a::calibration &secondary_cal,
+                                                        const Transformation &secondary_to_main)
+{
+    k4a::calibration cal = secondary_cal;
+    k4a_calibration_extrinsics_t &ex = cal.extrinsics[K4A_CALIBRATION_TYPE_DEPTH][K4A_CALIBRATION_TYPE_COLOR];
+    for (int i = 0; i < 3; ++i)
+    {
+        for (int j = 0; j < 3; ++j)
+        {
+            ex.rotation[i * 3 + j] = static_cast<float>(secondary_to_main.R(i, j));
+        }
+    }
+    for (int i = 0; i < 3; ++i)
+    {
+        ex.translation[i] = static_cast<float>(secondary_to_main.t[i]);
+    }
+    cal.color_camera_calibration = main_cal.color_camera_calibration;
+    return cal;
+}
+
+vector<float> calibration_to_color_camera_dist_coeffs(const k4a::calibration &cal)
+{
+    const k4a_calibration_intrinsic_parameters_t::_param &i = cal.color_camera_calibration.intrinsics.parameters.param;
+    return { i.k1, i.k2, i.p1, i.p2, i.k3, i.k4, i.k5, i.k6 };
+}
+
+// Takes the images by value so we can change them when shown.
+bool find_chessboard_corners_helper(const cv::Mat &main_color_image,
+                                    const cv::Mat &secondary_color_image,
+                                    const cv::Size &chessboard_pattern,
+                                    vector<cv::Point2f> &main_chessboard_corners,
+                                    vector<cv::Point2f> &secondary_chessboard_corners)
+{
+    bool found_chessboard_main = cv::findChessboardCorners(main_color_image,
+                                                           chessboard_pattern,
+                                                           main_chessboard_corners);
+    bool found_chessboard_secondary = cv::findChessboardCorners(secondary_color_image,
+                                                                chessboard_pattern,
+                                                                secondary_chessboard_corners);
+
+    // Cover the failure cases where chessboards were not found in one or both images.
+    if (!found_chessboard_main || !found_chessboard_secondary)
+    {
+        if (found_chessboard_main)
+        {
+            cout << "Could not find the chessboard corners in the secondary image. Trying again...\n";
```

Comment:
Should we be using the term chessboard or checkerboard? I don't really care which one, lets just be consistent throughout.

Reply:
Should be checkerboard for sure, as a chessboard is a very specific row x column configuration